### PR TITLE
Fix too long encoded word

### PR DESCRIPTION
**Pull request by TornaxO7**
This PR should fix https://github.com/soywod/rfc2047-decoder/issues/29

- This PR also adds more docs of the API
- There are some breaking changes in the API, for example the way how you set the parser-behaviour if it encounts an invalid encoded-word.
- You can now decide, how the parser should behave if an encoded word is incorrectly written:
  - Abort (the lexer will return an `Err`)
  - Decode (the lexer will just decode as if it would be a valid encoded word)
  - Skip (the lexer will treat it as a clear text)

### CODE REVIEW

**Comment by TornaxO7** [comment]
Sorry for the long break! This semester was (and still is) a bit rough, but I'm slowly getting some free time back.

File: src/decoder.rs [hunk 1]
```diff
@@ -3,38 +3,133 @@ use thiserror::Error;
...
 use crate::{evaluator, lexer, parser};
 
-#[derive(Error, Debug)]
+/// The possible errors which can occur while parsing the string.
+#[derive(Error, Debug, PartialEq)]
 pub enum Error {
+    /// Symbolises that an error occured in the lexer.
     #[error(transparent)]
-    Lexer(#[from] lexer::Error),
+    Lexer(#[from] lexer::LexerError),
```
Comment (soywod) [inline src/decoder.rs:11]: Either `lexer::Error` or `LexerError` instead of `lexer::LexerError`

File: src/decoder.rs [hunk 2]
```diff
@@ -94,12 +200,12 @@ mod tests {
...
 
         #[test]
-        fn test_example_6() {
+        fn example_6() {
             assert_eq!(decode("=?ISO-8859-1?Q?a_b?=").unwrap(), "a b");
         }
 
         #[test]
-        fn test_example_7() {
+        fn example_7() {
```
Comment (soywod) [inline src/decoder.rs:208]: Could be a good occasion to name those tests properly (with a more explicit name).

File: src/decoder.rs (hunk 1 above)
Comment (TornaxO7) [reply src/decoder.rs:11]: fixed

File: src/decoder.rs (hunk 2 above)
Comment (TornaxO7) [reply src/decoder.rs:208]: done

**Review by soywod — APPROVED** [review]

**Pull request merged**